src/main/java/net/projectzombie/region_rotation/modules/BaseState.java

```diff
@@ -171,7 +183,36 @@ else if (this.currentState.equals(swapState.getRegionName()))
         {
             return true;
         }
-
         return this.copyFrom(swapState, rotateAir);
     }
+
+    /**
+     * Used mostly for debug, it prints a nice and easy read of what the BaseState holds.
+     * @return Readable BaseState info.
+     */
+    public String toString()
+    {
+        String altText = ", Alts: ";
+        for (AltState alt : altStates.values())
+            altText += "- " + alt.getRegionName() + "@" + alt.getWorld().getName();
+        if (altText.equals(", Alts: "))
+            altText = ", No alts";
+        return "Main R: " + getRegionName() + "@" + getWorld().getName() +
```

Comment:
Use StringBuilder. When you add to a string using `+`, it takes the current string and new string, allocates memory the size of both string's lengths, then copies the strings. Very inefficient when you have lots of `+` strings. StringBuilder is similar to an ArrayList, it will double it's allocation size each time you reach capacity.